active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,592 @@
+3. When the `InputValidationKind` property is set to `Auto`, the control will use the value of the `ValidationCommand.InputValidationKind` property for the value. If both are `Auto` the default bevahior should be `Inline`.
+
+The markup compiler uses this interface to properly generate code when using x:Bind if and only if:
+1. The Source implements INotifyDataErrorInfo and INotifyPropertyChanged 
+2. The Target property is the one specified by the TargetType's InputPropertyAttribute
+3. The BindingMode is set to TwoWay
+
+The code generated provides the following functions:
+1. Generate code that listens to the Source's INotifyDataErrorInfo.ErrorsChanged event and populates the Target's 'IInputValidationControl.ValidationErrors' property.
+2. Generate and automatically set the ValidationContext property based on the Source property. If that property has any System.ComponentModel.DataAnnotations.ValidationAttributes, the associated property on the ValidationContext class will be set.
+
+         /// <summary>
+         /// Interface for Controls that participate in input validation.
+         /// <summary/>
+          [webhosthidden]
+         interface IInputValidationControl 
+         {
+             /// <summary>
+             /// Collection of errors to display based on the InputValidationKind property.
+            /// </summary>
+            Windows.Foundation.Collections.IObservableVector<Windows.UI.Xaml.Controls.InputValidationError> ValidationErrors{ get; };
+            /// <summary>
+            ///  Extra information about the Source of the validation. 
+            ///  Note: This is automatically set by the XAML markup compiler. See summary above. 
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationContext ValidationContext;
+            /// <summary>
+            ///  DataTemplate that expresses how the errors are displayed.
+            /// </summary>
+            Windows.UI.Xaml.DataTemplate ErrorTemplate;
+            /// <summary>
+            ///  Determines how the control should visualize validation errors.
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationKind InputValidationKind;
+            /// <summary>
+            ///  Determines how the ValidationCommand should perform validation.
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationMode InputValidationMode;
+            /// <summary>
+            /// Command associated with the IInputValidationControl. See InputValidationCommand for more
+            /// information.
+            /// </summary>
+            Windows.UI.Xaml.Controls.InputValidationCommand ValidationCommand;
+         };
+   
+## InputValidationCommand
```

Comment:
Did WPF have an equivalent of this? I know we got some pushback on having the framework define Commanding types in the framework layer -- I think folks were expecting these types to come out of the .NET layer. Now maybe .NET doesn't have any equivalent types that can stand in here, but I'd like to understand the need that this is satisfying. Or maybe just the "Command" part of the type name is throwing me off.

Reply:
no they don't, we're going to remove this type and come up with a different, and simpler model for achieving what it's purpose was